### PR TITLE
02WeekPigLatinExec

### PR DESCRIPTION
**Pull request by bireswarb**
## Checkpoint Rubric
This is the rubric that your instructor will use to grade your checkpoints. Please do not edit. 

**Checkpoint 1**
- [ ] All tests passed: 40 points
- [ ] Proper use of documentation (commenting on code): 15 points
- [ ] Properly indented code: 15 points
- [ ] Demonstrated effective use of JavaScript: 30 points

**Checkpoint 2**
- [ ] The application works as it should: 40 points
- [ ] Proper use of documentation (commenting on code): 15 points
- [ ] Properly indented code: 15 points
- [ ] Demonstrated effective use of JavaScript and the DOM API: 30 points

**Checkpoint 3**
- [ ] Use of React: 25 points
- [ ] Accesses an API: 25 points
- [ ] Proper use of documentation (commenting on code): 25 points
- [ ] The application functions as it should: 25 points

### CODE REVIEW

File: 02week/pigLatin.js [hunk 1]
```diff
@@ -8,9 +8,56 @@ const rl = readline.createInterface({
...
+      console.log('Input letters are:');
+    console.log(item);
+  }
+  let index;
+  let newStr1;
+  let newStr2;
+  let vowelFound = false;
+  for(index = 0; index<inputStr.length;index++){
+    switch(inputStr[index]){
+      case 'a':
```
Comment (reneemeyer) [inline 02week/pigLatin.js:27]: this is a really long switch statement and there's some redundancy, what's a better way to check if something is a vowel?

File: 02week/pigLatin.js [hunk 2]
```diff
@@ -8,9 +8,56 @@ const rl = readline.createInterface({
...
 
-  // Your code here
+  inputStr.forEach(isVowel);
+  function isVowel(item,index){
+    if(index == 0)
+      console.log('Input letters are:');
+    console.log(item);
+  }
+  let index;
+  let newStr1;
```
Comment (reneemeyer) [inline 02week/pigLatin.js:22]: you're using a lot of lets and opening the scope, what's a better way to do this?

File: 02week/pigLatin.js [hunk 3]
```diff
@@ -8,9 +8,56 @@ const rl = readline.createInterface({
...
 
 
-function pigLatin(word) {
+function pigLatin(input) {
+  let inputStr = input.split('');
+  let counter  = 0;
 
-  // Your code here
+  inputStr.forEach(isVowel);
+  function isVowel(item,index){
```
Comment (reneemeyer) [inline 02week/pigLatin.js:16]: use es6 function syntax

File: 02week/pigLatin.js [hunk 4]
```diff
@@ -8,9 +8,56 @@ const rl = readline.createInterface({
...
+      case 'O':
+      case 'U':
+           newStr1 = inputStr.slice(index);
+           newStr2 = inputStr.slice(0,index);
+           newStr1=newStr1.join('');
+           newStr2=newStr2.join('');
+           vowelFound = true;
+       break;
+    }
+    if(vowelFound)
```
Comment (reneemeyer) [inline 02week/pigLatin.js:44]: why are you looking if a vowel is found here and on line 48?

**Review by reneemeyer — CHANGES_REQUESTED** [review]
Good start, you're logic is correct but you are using a lot of unnecessary methods and the code is difficult to follow. I would refactor to increase readability and reduce redundancy.

File: 02week/pigLatin.js (hunk 3 above)
Comment (bireswarb) [reply 02week/pigLatin.js:16]: Pushed new changes with refactored code

File: 02week/pigLatin.js (hunk 2 above)
Comment (bireswarb) [reply 02week/pigLatin.js:22]: Pushed new changes with refactored code

File: 02week/pigLatin.js (hunk 1 above)
Comment (bireswarb) [reply 02week/pigLatin.js:27]: Pushed new changes with refactored code

File: 02week/pigLatin.js (hunk 4 above)
Comment (bireswarb) [reply 02week/pigLatin.js:44]: Pushed new changes with refactored code

File: 02week/pigLatin.js (hunk 3 above)
Comment (bireswarb) [reply 02week/pigLatin.js:16]: https://github.com/bireswarb/javascript-workbook/pull/7

File: 02week/pigLatin.js (hunk 2 above)
Comment (bireswarb) [reply 02week/pigLatin.js:22]: https://github.com/bireswarb/javascript-workbook/pull/7

File: 02week/pigLatin.js (hunk 1 above)
Comment (bireswarb) [reply 02week/pigLatin.js:27]: https://github.com/bireswarb/javascript-workbook/pull/7

File: 02week/pigLatin.js (hunk 4 above)
Comment (bireswarb) [reply 02week/pigLatin.js:44]: https://github.com/bireswarb/javascript-workbook/pull/7

**Pull request merged**